Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift

```diff
@@ -525,3 +538,13 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
         documentAttachments.append(contentsOf: documentsInfo)
     }
 }
+
+public extension MessageComposerViewController.Delegate {
+    static func wrap<T: MessageComposerViewControllerDelegate>(
```

Comment:
Could you please add documentation to this method? :) Like what `wrap` means in this context. A bit confused with the code 😅 And why is it needed/explain the concept thanks :)

Reply:
Because most of our types are generics on `ExtraData`, their delegate needs to have `ExtraData` associated type.
This limit protocol usage (now it can be used only as generic constraint). To store reference on delegate, we need to type erase it.
Wrapping delegate in struct with closures is one of ways to do it. (exactly same approach is used in `ChatMessageListVC`.

I'm not sure documenting here all of that will be helpful, because type erasure problem is all around us (take a look how `WhateverController` in LLC stores their delegates.
I would expect this to be separate chapter in our library documentation. 
cc @VojtaStavik